src/lsp.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  ServerOptions,
+  Executable,
+  TransportKind
+} from "vscode-languageclient/node";
+
+import {
+  ExtensionContext,
+  workspace,
+  WorkspaceConfiguration,
+} from "vscode"
+
+import {
+  findLanguageServer,
+} from "./utils";
+
+import * as which from "which"
+
+export class MesonLanguageClient {
+  config: WorkspaceConfiguration
+  ls: LanguageClient | null = null
+
+  constructor(_context: ExtensionContext) {
+    this.config = workspace.getConfiguration("mesonbuild");
+    let serverModule = this.languageServerPath;
+
+    if (serverModule == null)
+      return;
+    this.startLanguageServer(serverModule);
+  }
+
+  startLanguageServer(serverModule: string): void {
+    let clientOptions: LanguageClientOptions = {
```

Comment:
Seems like there could be more "const" instead of "let" here, but I am definitely no expert :).